File: DESAFIO.BLOG.Domain/Entities/ApplicationUser.cs
using Microsoft.AspNetCore.Identity;

namespace DESAFIO.BLOG.Domain.Entities
{
    public class ApplicationUser : IdentityUser<Guid>
    {
        public string Name { get; set; }
        public string JwtToken { get; set; }
        public ICollection<Post> Posts { get; set; }
        public ICollection<ChatMessage> ChatMessagesSent { get; set; } // Mensagens enviadas
        public ICollection<ChatMessage> ChatMessagesReceived { get; set; } // Mensagens recebidas
    }
}

File: Front.Angular/src/app/Models/ApplicationUser.ts
import { ChatMessage } from "./ChatMessage";
import { Post } from "./Post";

export interface ApplicationUser {
  id: string;
  userName: string;
  email: string;
  emailConfirmed: boolean;
  phoneNumber: string;
  phoneNumberConfirmed: boolean;
  twoFactorEnabled: boolean;
  lockoutEnd: Date;
  lockoutEnabled: boolean;
  accessFailedCount: number;
  name: string;
  jwtToken: string;
  posts: Post[];
  chatMessagesSent: ChatMessage[];
  chatMessagesReceived: ChatMessage[];
}
